airflow/migrations/versions/0137_2_9_0_add_priority_weight_strategy_to_task.py

```diff
@@ -0,0 +1,47 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""add priority_weight_strategy to task_instance
+
+Revision ID: 624ecf3b6a5e
+Revises: 1fd565369930
+Create Date: 2023-10-29 02:01:34.774596
+
+"""
+
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "624ecf3b6a5e"
+down_revision = "ab34f260b71c"
+branch_labels = None
+depends_on = None
+airflow_version = "2.9.0"
+
+
+def upgrade():
+    """Apply add priority_weight_strategy to task_instance"""
+    with op.batch_alter_table("task_instance") as batch_op:
+        batch_op.add_column(sa.Column("_priority_weight_strategy", sa.JSON()))
```

Comment:
I understand we need to store some context information about the selcted priority weight strategy - but do we really need to add this to the DB? TaskInstance is the most largest table in the DB scheme and potentially contains millions of rows. Do we really want to store the same values in mostly millions of cases? Or can we leave it `NULL` and store only a value if we have this special rule and data needs stored?

I like the approach of this PR in general but fear this will create a lot of overhead in DB. Especially as it is a JSON field.